iree/compiler/Conversion/Common/LinalgBufferizePass.cpp

```diff
@@ -624,6 +626,16 @@ LogicalResult convertInterfaceStoreTensorOp(
   return success();
 }
 
+// Forwards buffer assigned to cast inputs to its outputs.
+LogicalResult convertTensorCastOp(OpBuilder &b,
+                                  WorkgroupMemoryAllocationFn allocationFn,
+                                  tensor::CastOp castOp,
+                                  BlockAndValueMapping &bvm) {
+  Value inputBuffer = bvm.lookup(castOp.source());
+  bvm.map(castOp.dest(), inputBuffer);
```

Comment:
It would be good to note that this is missing a fallback of allocating a buffer and doing a copy (since `tensor.cast` has copy semantics).